restructure projects ui

nayasinghania opened the pull request:
Worked with @evanugarte to remake the projects ui with each project card being in a separate row and the icons being removed. 
please leave any suggestions below.
also lmk if i should add the more recent projects (issue #1525)

![image](https://github.com/user-attachments/assets/95de25a4-f3b5-4684-8928-02cadd05e344)

nayasinghania committed:
restructure projects ui

joshua-demo commented on src/Pages/Projects/Projects.js:
```diff
@@ -1,201 +1,55 @@
-      ),
-    },
-  ];
+import ProjectCard from './Components/ProjectCard';
```
can put all the import statements at the top of the file

nayasinghania replied:
oops yeah my bad I'll move it

joshua-demo commented on src/Pages/Projects/Projects.js:
```diff
@@ -1,201 +1,55 @@
-            </article>
-          ))}
-        </div>
+    <div className="dark:bg-gray-900">
```
I think we will need margins on the left and right for mobile view

nayasinghania replied:
ok will add

steeevin88 commented:
agree with josh's comments, if you could add [SCEta](https://sce.sjsu.edu/transit/#10th-and-taylor) to the page it would be really cool 😄 

i personally dislike having the pancake printer + RFID projects because they were never finished, I added them ~2 years ago when we didn't have that many other projects. maybe we could add them? (ex. Cleezy, Messaging)

weslayer commented:
nice work bro, it would be very cool to add recent projects 😎😎

nayasinghania commented:
> agree with josh's comments, if you could add [SCEta](https://sce.sjsu.edu/transit/#10th-and-taylor) to the page it would be really cool 😄 
> 
> i personally dislike having the pancake printer + RFID projects because they were never finished, I added them ~2 years ago when we didn't have that many other projects. maybe we could add them? (ex. Cleezy, Messaging)

Ok, I can remove the pancake printer and RFID projects and add the newer completed ones

nayasinghania committed:
Update Projects.js

nayasinghania committed:
mobile layout fixes + added new projects

nayasinghania committed:
Merge branch 'restructure-projects-ui' of https://github.com/SCE-Development/Clark into restructure-projects-ui

nayasinghania committed:
remove extra space

nayasinghania commented:
@joshua-demo @steeevin88 I fixed the layout and padding for mobile and added the new projects. Wasn't entirely sure about the details of the new projects added, so suggestions of what to put would helpful. Also there is a placeholder image for Cleezy bc i cant access it to take a screenshot.
![Screenshot 2024-10-18 123535](https://github.com/user-attachments/assets/a0b63ff2-5bc9-4cd8-9af7-ab55de7f5bb2)

![Screenshot 2024-10-18 123553](https://github.com/user-attachments/assets/4d7a700f-f27f-4505-8c58-37316877545f)

evanugarte commented on public/images/messaging.png:
can we upload these to github's cdn instead?

https://gist.github.com/NawalJAhmed/2168f7659c08b6a033e7f6daf8db69a6

nayasinghania replied:
done

evanugarte commented on src/Pages/Projects/Projects.js:
```diff
@@ -1,201 +1,75 @@
+    'information': 'FastAPI',
+    'caption': 'A url shortening service created by SCE'
+  },
+  /* {
+    'link': 'https://github.com/SCE-Development/RFID-door-lock',
+    'image': 'https://user-images.githubusercontent.com/59713070/235862570-70f92c0a-8e18-4ddf-b7c8-bdb21723480f.jpeg',
+    'name': 'RFID Door System',
+    'information': 'Firmware with Arduino',
+    'caption': 'SCE\'s development team officers created an RFID card reader and door control relay to allow entry into the clubroom\'s office using a Clipper card.'
+  },
+  {
+    'link': 'https://github.com/SCE-Development/',
+    'image': 'https://user-images.githubusercontent.com/59713070/235861717-5315d9df-fb5e-4414-84bb-0334fe769271.jpeg',
+    'name': '3D Pancake Printer',
+    'information': '3D Printing with Marlin Firmware',
+    'caption': 'In the spirit of our love for pancakes, SCE\'s hardware team constructed a 3D Pancake Printer. Does it work? No. When completed it will create intricate pancake designs.'
```
lets remove the commented out code

nayasinghania replied:
done

evanugarte commented on src/Pages/Projects/Projects.js:
```diff
@@ -1,201 +1,75 @@
+        <br />
       </div>
-    </section>
+      {projects.map((project, index) => (
+        <>
+          <ProjectCard key={index} project={project} />
```
see link for explanation on avoiding lists and keys
https://legacy.reactjs.org/docs/lists-and-keys.html#keys

```suggestion
      {projects.map((project) => (
        <>
          <ProjectCard key={project.link} project={project} />
```

nayasinghania replied:
fixed

nayasinghania committed:
remove commented out code

nayasinghania committed:
fix list/key issue

nayasinghania committed:
move images to github cdn

steeevin88 commented:
https://github.com/user-attachments/assets/de1017ac-ca79-46e5-b3b4-5e62666713fb

here's a picture for Cleezy

everything else LGTM, nice work 😃

evanugarte commented on src/Pages/Projects/Components/ProjectCard.js:
```diff
@@ -1,19 +1,36 @@
 
-
-export default function ProjectCard({link, image, name, subnote, caption}) {
+export default function ProjectCard({project}) {
```
```suggestion
export default function ProjectCard(project) {
```

evanugarte commented on src/Pages/Projects/Projects.js:
```diff
@@ -1,201 +1,61 @@
-    </section>
+      {projects.map((project) => (
+        <>
+          <ProjectCard key={project.name} project={project} />
```
this will pass each project field in as its own prop, see the other comment in ProjectCard.js
```suggestion
          <ProjectCard key={project.name} {...project} />
```

nayasinghania committed:
Update src/Pages/Projects/Components/ProjectCard.js

Co-authored-by: Evan Ugarte <36345325+evanugarte@users.noreply.github.com>

nayasinghania committed:
Update src/Pages/Projects/Projects.js

Co-authored-by: Evan Ugarte <36345325+evanugarte@users.noreply.github.com>

nayasinghania committed:
Update Projects.js
